### PR TITLE
migrations: update morph dependecy and apply changes

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Bump morph dependency and reflect the changes in the API.

We dropped the driver config and all the config will be passed from the morph engine.

### CODE REVIEW

**Comment by wiggin77** [comment]
CI error here:
```
2022-11-15T21:22:08.3506037Z === RUN   TestDeletedMembershipBoardsMigration
2022-11-15T21:22:08.3506365Z === RUN   TestDeletedMembershipBoardsMigration/should_detect_a_board_linked_to_a_team_in_which_the_owner_has_a_deleted_membership_and_restore_it
2022-11-15T21:22:08.3507883Z     foundation.go:34: error setting up the migrator: driver: mysql, message: failed while executing query, command: create_migrations_table_if_not_exists, originalError: Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(Version bigint(20) NOT NULL, Name varchar(64) NOT NULL, PRIMARY KEY (Version)) ' at line 1, query: 
2022-11-15T21:22:08.3507987Z         
2022-11-15T21:22:08.3508559Z         CREATE TABLE IF NOT EXISTS  (Version bigint(20) NOT NULL, Name varchar(64) NOT NULL, PRIMARY KEY (Version)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4
```

**Comment by isacikgoz** [comment]
@wiggin77 addressed the issue, it was an error on my end. Tests are passing now.

File: server/services/store/sqlstore/migrationstests/boards_migrator_test.go [hunk 1]
```diff
@@ -65,12 +65,16 @@ func (bm *BoardsMigrator) runMattermostMigrations() error {
...
 
-	driver, err := bm.getDriver("")
+	driver, err := bm.getDriver()
 	if err != nil {
 		return err
 	}
 
-	engine, err := morph.New(context.Background(), driver, src)
+	options := []morph.EngineOption{
+		morph.SetStatementTimeoutInSeconds(1000000),
```
Comment (wiggin77) [inline server/services/store/sqlstore/migrationstests/boards_migrator_test.go:74]: Statement timeout is 11 days?

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: server/services/store/sqlstore/migrationstests/boards_migrator_test.go (hunk 1 above)
Comment (isacikgoz) [reply server/services/store/sqlstore/migrationstests/boards_migrator_test.go:74]: Yeah, the default value we set was 600 seconds, which is quite small for big DBs. I think we can maybe add a constant to be a year practically being infinite.

**Review by mgdelacroix — APPROVED** [review]

**Pull request merged**